### PR TITLE
Add documentation for custom CSPM rules

### PR DESCRIPTION
**Pull request by danieldebeer-dd**
### What does this PR do?
Adds documentation for custom CSPM rules

### Motivation
To help customers understand custom CSPM rules

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by PronoyC — COMMENTED** [review]
will review and approve after the docs team has made changes

**Comment by kayayarai** [comment]
Tracking in https://datadoghq.atlassian.net/browse/DOCS-4041

**Comment by hestonhoffman** [comment]
Weird that it didn't cause hiccups in the preview build, but here's an example of a file where it failed: https://docs-staging.datadoghq.com/brian.deutsch/shortcodes/integrations/slack/

File: content/en/security_platform/cspm/custom_rules/writing_rego_rules.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    resource = input.some_resource_type[_]
+    result := dd_output.format(resource,eval(resource))
+}
+```
+
+Take a close look at each part of this module to understand how it works.
+
+### Import statements
+
+The first line contains the declaration `package datadog`. A [package][4] groups Rego modules into a single namespace, allowing modules to be imported safely. Currently, importing user modules is not a feature of custom rules. All posture management rules are grouped under the `datadog` namespace. For your results to be returned properly, you should also group your rules under the `package datadog` namespace.
```
Comment (kayayarai) [inline content/en/security_platform/cspm/custom_rules/writing_rego_rules.md:54]:
```suggestion
The first line contains the declaration `package datadog`. A [package][4] groups Rego modules into a single namespace, allowing modules to be imported safely. Currently, importing user modules is not a feature of custom rules. All posture management rules are grouped under the `datadog` namespace. For your results to be returned properly, group your rules under the `package datadog` namespace. 
```

**Review by kayayarai — APPROVED** [review]

**Comment by kayayarai** [comment]
I have moved all this content to the greater CSM PR https://github.com/DataDog/documentation/pull/15439 Any further review or change should happen on it instead.

**Pull request closed without merging**